src/coreclr/vm/threads.cpp

```diff
@@ -1562,7 +1588,10 @@ Thread::Thread()
     m_ioThreadPoolCompletionCount = 0;
     m_monitorLockContentionCount = 0;
 
-    InitContext();
+    {
+        GCX_COOP_NO_THREAD_BROKEN();
+        m_pDomain = SystemDomain::System()->DefaultDomain();
+    }
```

Comment:
```suggestion
    m_pDomain = SystemDomain::System()->DefaultDomain();
```
This `GCX_COOP_NO_THREAD_BROKEN` should not be needed. Everything in `SystemDomain::System()->DefaultDomain()` are `LIMITED_METHOD_DAC_CONTRACT` methods (ie they don't do anything interesting)

Reply:
Will do.